src/Frame.h

```diff
@@ -170,12 +169,13 @@ class Frame {
         const CARTA::MomentRequest& moment_request, CARTA::MomentResponse& moment_response,
         std::vector<carta::CollapseResult>& collapse_results);
     void StopMomentCalc();
-    void IncreaseMomentsCount();
-    void DecreaseMomentsCount();
 
     // Save as a new file or convert it between CASA/FITS formats
     void SaveFile(const std::string& root_folder, const CARTA::SaveFile& save_file_msg, CARTA::SaveFileAck& save_file_ack);
 
+    // Use a shared lock for long time calculations, use an exclusive lock for the object destruction
+    mutable std::shared_mutex _life_mutex;
```

Comment:
if this is public, it shouldn't be `_life_mutex`. Either make it public, or (probably neater) encapsulate the shared and exclusive locks in functions.